src/validation.ts

```diff
@@ -0,0 +1,76 @@
+import addFormats from 'ajv-formats';
+import Ajv, { ValidateFunction } from 'ajv';
+
+import workflowStepTemplate from './workflow_step_template_schema.json';
+import workflowTemplate from './workflow_template_schema.json';
+
+interface Options {
+  schema?: 'workflowTemplate' | 'workflowStepTemplate';
+}
+
+interface Validation {
+  schema: ValidateFunction['schema'];
+  errors: ValidateFunction['errors'];
+}
+
+const ajv = new Ajv({
+  // Check all rules collecting all errors.
+  allErrors: true,
+  // Allow using multiple non-null types in "type" keyword
+  allowUnionTypes: true,
+  // Include reference to the part of the schema and validated data in errors.
+  verbose: true,
+  // Custom keywords that are not part of the JSON schema draft
+  keywords: ['columns', 'rules'],
```

Comment:
where does this list came from, and why do they need to be provided?

Reply:
[here](https://github.com/labforward/laboperator-workflow-schema/blob/122d06f2eb0059eb4587100d35a65f3043e2b963/dist/workflow_template_schema.json#L2280) and [here](https://github.com/labforward/laboperator-workflow-schema/blob/122d06f2eb0059eb4587100d35a65f3043e2b963/dist/workflow_template_schema.json#L2370), those are not valid keywords according to the JSON draft schema. We either have to turn off strict validation or add these in order to prevent compilation errors.

Reply:
Actually, lemme try if we can change the schema and our templates are still valid. I will report back.

Reply:
Fixed